modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/CoreService.java

```diff
@@ -75,75 +75,57 @@ public void deactivate(ComponentContext context) {
     }
 
     @Override
-    public void registerContribution(Object contrib, String point, ComponentInstance contributor) {
-        if (VERSION_REMOVAL_POLICY_XP.equals(point)) {
-            registerVersionRemovalPolicy((CoreServicePolicyDescriptor) contrib);
-        } else if (ORPHAN_VERSION_REMOVAL_FILTER_XP.equals(point)) {
-            registerOrphanVersionRemovalFilter((CoreServiceOrphanVersionRemovalFilterDescriptor) contrib);
-        } else {
-            throw new RuntimeException("Unknown extension point: " + point);
-        }
+    public void start(ComponentContext context) {
+        versionRemovalPolicy = this.<CoreServicePolicyDescriptor> getRegistryContribution(VERSION_REMOVAL_POLICY_XP)
+                                   .map(this::createVersionRemovalPolicy)
+                                   .orElse(null);
```

Comment:
What do you think about giving `DEFAULT_VERSION_REMOVAL_POLICY` to `orElse` and removing the ternary in `getVersionRemovalPolicy`?